src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -539,12 +554,32 @@ public with sharing class PMT_PaymentWizard_CTRL {
 
     /*******************************************************************************************************
     * @description Returns true if the user has permissions to access, create and delete npe01__oppPayment__c
+    * and all relevant fields are both readable and modifiable
     * @return Boolean
     */
     private Boolean hasPaymentAccess() {
-        Schema.DescribeSObjectResult paymentDescribe = npe01__oppPayment__c.sObjectType.getDescribe();
+        Boolean accessResult = false;
+
+        SObjectType pmt = npe01__OppPayment__c.getSObjectType();
+        if (perms.canCreate(pmt) && UTIL_Permissions.canDelete(Schema.SObjectType.npe01__oppPayment__c.getName(), false)) {
+            Set<SObjectField> sObjectFields = new Set<SObjectField>{
+                npe01__oppPayment__c.fields.npe01__check_reference_number__c,
+                npe01__oppPayment__c.fields.npe01__custom_payment_field__c,
+                npe01__oppPayment__c.fields.npe01__paid__c,
+                npe01__oppPayment__c.fields.payment_acknowledged_date__c,
```

Comment:
I don't think these two PaymentAcknowledge fields are required either. They're not referenced by the page or controller at all. Overall, the permissions check should be limited to only fields that are directly manipulated by the VF page and/or this controller. Otherwise we risk breaking the page for customers who haven't granted access to fields that aren't needed by the Schedule or Writeoff Wizard pages.